components/prism-solidity.js

```diff
@@ -0,0 +1,77 @@
+Prism.languages.solidity = Prism.languages.extend('clike', {
+    'keyword': [
+		{pattern: /\b(?:var|import|function|constant|view|pure|payable|storage|memory|if|else|for|while|do|break|continue|returns?|private|public|internal|external|inherited|this|suicide|selfdestruct|emit|new|is|throw|revert|assert|require|\_)\b/},
+        {
+            pattern: /\b(contract|interface|library|using|struct|function|modifier)\s+([A-Za-z_]\w*)(?:\s+is\s+((?:[A-Za-z_][\,\s]*)*))?\b/,
```

Comment:
Please use non-capturing groups `(?:)` everywhere capture is not needed. Here, all groups can be made non-capturing.
Also, I think there are unneeded groups in the second part of the regexp:
- `([A-Za-z_]\w*)` I think the parentheses here can be removed
- `((?:[A-Za-z_][\,\s]*)*)` the outer parentheses here can be removed too, and you don't need to escape `,`.